### DISCUSSION TITLE
want to connect chatwoot to whatsapp cloud api but its asking api key. where to get them

### DISCUSSION THREAD

**Discussion by chaitanysharan** [chatwoot/chatwoot#5102] [Chatwoot APIs]
I am trying to integrate official whatsapp cloud API. But when I try to connect, it asks for API KEY and WEBHOOK VERIFY TOKEN. where can I get them.

<img width="1920" alt="Screenshot 2022-07-25 at 12 47 46 PM" src="https://user-images.githubusercontent.com/8451508/180720312-282cab26-68d6-48be-9ba5-f2008ad2a7e4.png">

**Comment by sojan-official** [comment]
please refer https://www.chatwoot.com/docs/product/channels/whatsapp/whatsapp-cloud

**Comment by chaitanysharan** [comment]
I have read this document already, but I don't know where I can get these 2 data. I could only find phone number ID and Business Account ID.
Please enlighten me
where should I go to get these data

**Reply by sojan-official** [reply]
api token -> https://www.chatwoot.com/docs/product/channels/whatsapp/whatsapp-cloud#configure-a-system-user-for-your-facebook-bussiness-manager ( you need to generate a system user in Facebook business manager following the docs )

verify token -> you can choose any string / text you wish https://www.chatwoot.com/docs/product/channels/whatsapp/whatsapp-cloud#configure-webhook-in-facebook-developer-app

**Comment by chaitanysharan** [comment]
Hi, I could manage to get my account up and running. Now I have another
question. How can I send a template message with a media attachment? I have
to send a customer a template message with his invoice attached to the
message. So every time I send a message the attachment file will be
different.
please advise
Chaitany Sharan
